include/lsst/afw/detection/Footprint.h

```diff
@@ -340,212 +235,19 @@ class Footprint : public lsst::daf::base::Citizen,
 
     //@{
     /// Persistence implementation functions made available for derived classes
-    void readSpans(afw::table::BaseCatalog const & spanCat);
-    void readPeaks(afw::table::BaseCatalog const & peakCat);
+    void readSpans(table::BaseCatalog const & spanCat);
+    void readPeaks(table::BaseCatalog const & peakCat);
     //@}
 
     friend class FootprintFactory;
 
-private:
-
-    friend class FootprintMerge;
+ private:
 
-    static int id;
-    mutable int _fid;                    //!< unique ID
-    size_t _area;                           //!< number of pixels in this Footprint (not the area of the bbox)
+    friend class FootprintMerge; // Maybe make it not a friend
 
-    SpanList _spans;                     //!< the Spans contained in this Footprint
-    geom::Box2I _bbox;                   //!< the Footprint's bounding box
-    PeakCatalog _peaks;                     //!< the Peaks lying in this footprint
-    mutable geom::Box2I _region;         //!< The corners of the MaskedImage the footprints live in
-    bool _normalized;                    //!< Are the spans sorted?
+    std::shared_ptr<SpanSet> _spans;    //!< The SpanSet representing area on image
+    PeakCatalog _peaks;                 //!< The peaks lying in this footprint
+    geom::Box2I _region;     //!< The corners of the MaskedImage the footprints live in
 };
 
-/**
- Given a vector of Footprints, fills the output "argmin" and "dist"
- images to contain the Manhattan distance to the nearest footprint (in
- "dist") and the identity of the nearest footprint (in "argmin").
-
- For example, if there are two footprints at y=0 covering x=[1,2] and [7,7],
-
- Index     :  0 1 2 3 4 5 6 7
-
- Footprints:  . 0 0 . . . . 1
-
- Argmin    :  0 0 0 0 0 1 1 1
-
- Dist      :  1 0 0 1 2 2 1 0
-
- "argmin" gives the index of the nearest footprint, and "dist" its
- Manhattan (L_1 norm) distance.  The pixel at index 4 is closest to
- footprint 0, and its distance is 2.
- */
-void nearestFootprint(std::vector<PTR(Footprint)> const& foots,
-                      lsst::afw::image::Image<std::uint16_t>::Ptr argmin,
-                      lsst::afw::image::Image<std::uint16_t>::Ptr dist);
```

Comment:
Does this function have any current users?  Like `clipToNonzero`, it seems like a fairly specialized algorithm that should perhaps live with the code that uses it rather than `Footprint` itself, but it also seems tricky enough to implement that we shouldn't remove it lightly.